application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/SpamSolrEntityMetadataExtractor.java

```diff
@@ -45,22 +50,16 @@ public class SpamSolrEntityMetadataExtractor implements SolrEntityMetadataExtrac
     private Logger logger;
 
     @Inject
-    private Provider<AdminToolsConfiguration> configurationProvider;
-
-    @Inject
-    private Provider<XWikiContext> wikiContextProvider;
+    @Named("current")
+    private DocumentReferenceResolver<String> documentReferenceResolver;
 
     @Override
     public boolean extract(XWikiDocument entity, SolrInputDocument solrDocument)
     {
         try {
-            Vector<BaseObject> results = entity.getObjects("XWiki.XWikiComments");
-            if (results != null && results.size() > configurationProvider.get().getSpamSize()) {
-                // add spammed field
-                solrDocument.addField("spammed", "true");
-            } else {
-                solrDocument.addField("spammed", "false");
-            }
+            List<BaseObject> results = entity.getXObjects(documentReferenceResolver.resolve("XWiki.XWikiComments",
+                entity.getDocumentReference().getWikiReference()));
```

Comment:
You should have a `public static final EntityReference COMMENTSCLASS_REFERENCE = new LocalDocumentReference("XWiki", "XWikiComments");` and use that in `getXObjects(COMMENTSCLASS_REFERENCE)`